### PR TITLE
feat: bring asset response type into shared package

### LINKED ISSUE #1335: [Subtask] - Share types for Asset
Share types for Testnet Wallet backend and Testnet Boutique backend
Controllers:

asset

### PR DESCRIPTION
**Pull request by rico191013**
### Context

- fixes #1335

### CODE REVIEW

File: packages/wallet/shared/src/types/asset.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export type Asset = {
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/asset.ts:1]: are these types used in backend as well?

Comment (beniaminmunteanu) [reply packages/wallet/shared/src/types/asset.ts:1]: #1351 
 The Types shared here should be imported in the backend to be used as the response type of controllers, 
and also in the frontend in order to import and not duplicate response types.

So i believe these types should also be added to the controllers?

**Review by beniaminmunteanu — DISMISSED** [review]

File: packages/wallet/shared/src/types/asset.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export interface Asset {
+  id: string
+  code: string
+  scale: number
+  withdrawalThreshold?: bigint
+  createdAt: string
+}
+
+export interface Rates {
```
Comment (dragosp1011) [inline packages/wallet/shared/src/types/asset.ts:9]: move rates type in a separate file

**Comment by beniaminmunteanu** [comment]
Keeping in mind the other places that need to have the types moved to shared: #1351 

My suggestion would be to have the interfaces in shared have a conventional name, like including "Response" in it such as "NameResponse"

So that in the "account" implementation for example we would have the "Account" type being used for the bussiness logic inside the backend, and that is not shared, but for the controller response type we would have "AccountResponse"

Wdyt?

**Review by dragosp1011 — DISMISSED** [review]

**Review by dragosp1011 — APPROVED** [review]

**Pull request merged**